src/agent/onefuzz/src/telemetry.rs

```diff
@@ -341,25 +341,26 @@ macro_rules! event {
 #[macro_export]
 macro_rules! log {
     ($level: expr, $msg: expr) => {{
-        use appinsights::telemetry::SeverityLevel;
-        use SeverityLevel::*;
-
-        {
-            let log_level = match $level {
-                SeverityLevel::Verbose => log::Level::Debug,
-                SeverityLevel::Information => log::Level::Info,
-                SeverityLevel::Warning => log::Level::Warn,
-                SeverityLevel::Error => log::Level::Error,
-                SeverityLevel::Critical => log::Level::Error,
-            };
-
-            let log_msg = $msg.to_string();
+        use appinsights::telemetry::SeverityLevel::{
+            Critical, Error, Information, Verbose, Warning,
+        };
 
-            log::log!(log_level, "{}", log_msg)
-        }
+        let log_level = match $level {
+            Verbose => log::Level::Debug,
+            Information => log::Level::Info,
+            Warning => log::Level::Warn,
+            Error => log::Level::Error,
+            Critical => log::Level::Error,
+        };
 
-        if let Some(client) = $crate::telemetry::client($crate::telemetry::ClientType::Instance) {
-            client.track_trace($msg, $level);
+        // while log::log will filter based on log level, the telemetry
+        // client does *not*.
+        if log_level <= log::max_level() {
```

Comment:
My only uncertainty with this is how it interacts with module-specific `RUST_LOG` settings like `RUST_LOG=onefuzz_supervisor=trace`. During dev, I almost never set e.g. `RUST_LOG=trace`, because that makes us emit log messages from dependencies.

I think this is fine for now, but it would be nice to be able to tune the telemetry log level separately.

Reply:
I think the next step would be to move to a more configurable logging mechanism, perhaps [fern](https://crates.io/crates/fern).